src/Symfony/Component/Routing/Loader/Configurator/RoutingConfigurator.php

```diff
@@ -59,4 +61,32 @@ public function __construct(RouteCollection $collection, PhpFileLoader $loader,
     {
         return new CollectionConfigurator($this->collection, $name);
     }
+
+    /**
+     * @return RouteCollection|RouteCollection[]
+     */
+    private function load($resource, $type = null, $ignoreErrors = false)
+    {
+        if ($this->loader instanceof FileLoader) {
+            if (null !== $this->path) {
+                $this->loader->setCurrentDir(\dirname($this->path));
+            }
+
+            return $this->loader->import($resource, $type, $ignoreErrors, $this->file);
```

Comment:
I wish we could avoid downcasting here. `PhpFileLoader` does `$this->setCurrentDir(\dirname($path));` internally, so this could be probably omitted. But the `$ignoreErrors` functionality will be lost.